.github/workflows/submodule-pin-check.yml

```diff
@@ -1,21 +1,48 @@
-name: Submodule Pin Check
+name: Merge Checks
 
 on:
   pull_request:
     branches: [ master ]
     types: [synchronize, opened, reopened]
 
+permissions:
+  statuses: write
+
 jobs:
   submodule-pin-check:
-    name: Submodule Pin Check
+    name: Check Submodule Pin
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
-          submodules: recursive
+          submodules: true
```

Comment:
I think there was a reason I had set this to recursive. Does this still work correctly without it?

Reply:
I tested changing go-ethereum and letting this submodules field set to true and it worked :)